src/libdrakvuf/json-profile.c

```diff
@@ -268,7 +268,53 @@ bool drakvuf_get_kernel_symbol_rva(drakvuf_t drakvuf,
     const char* function,
     addr_t* rva)
 {
-    return VMI_SUCCESS == vmi_get_symbol_addr_from_json(drakvuf->vmi, vmi_get_kernel_json(drakvuf->vmi), function, rva);
+    json_object* kernel_json = vmi_get_kernel_json(drakvuf->vmi);
+
+    if (drakvuf->os == VMI_OS_WINDOWS)
+        return VMI_SUCCESS == vmi_get_symbol_addr_from_json(drakvuf->vmi, kernel_json, function, rva);
+    else if (drakvuf->os == VMI_OS_LINUX)
+    {
+        if (VMI_FAILURE == vmi_get_symbol_addr_from_json(drakvuf->vmi, kernel_json, function, rva))
+        {
+            bool find = false;
+            for (uint8_t i = 0; i < 255; i++)
+            {
+                char tmp[64];
+                snprintf(tmp, sizeof(tmp), "%s.isra.%d", function, i);
+                if (VMI_SUCCESS == vmi_get_symbol_addr_from_json(drakvuf->vmi, kernel_json, tmp, rva))
+                {
+                    find = true;
+                    break;
+                }
+            }
+            if (!find)
+                return false;
+        }
+    }
```

Comment:
What if it's another OS?

Reply:
Change if/else to switch/case. I have not tested drakvuf under FreeBSD, but if a similar situation arises in the future, as with Linux, then it can be easily corrected.

Reply:
What I was pointing out is that you are returning true even if neither of the currently supported OS' were detected. Also, the os specific libdrakvuf functions ought to be abstracted into the os wrapper interface, just like all the other os specific functions.

Reply:
@tklengyel As far as I understand, you want to see something like this? (Changes in the last commit)

Reply:
Yes, take a look at https://github.com/tklengyel/drakvuf/blob/master/src/libdrakvuf/os.h, any new OS specific functionality should be setup like these.